packages/account/src/predicate/predicate.ts

```diff
@@ -28,39 +28,52 @@ import type {
 
 import { getPredicateRoot } from './utils';
 
+export type PredicateParams<T = InputValue[]> = {
+  bytecode: BytesLike;
+  provider: Provider;
+  abi?: JsonAbi;
+  inputData?: T;
+  configurableConstants?: { [name: string]: unknown };
+};
+
 /**
  * `Predicate` provides methods to populate transaction data with predicate information and sending transactions with them.
  */
-export class Predicate<ARGS extends InputValue[]> extends Account {
+export class Predicate<TInputData extends InputValue[]> extends Account {
   bytes: Uint8Array;
   predicateData: Uint8Array = Uint8Array.from([]);
-  predicateArgs: ARGS = [] as unknown as ARGS;
+  predicateArgs: TInputData = [] as unknown as TInputData;
```

Comment:
IMO this is confusing because they aren't really describing their function correctly. We should probably have `predicateData` and `predicateDataBytes`.

Reply:
Good point. I've renamed the variables but I noticed that `predicateData`, now named `predicateDataBytes` is never actually _set_ anywhere. 👁️

Reply:
@Dhaiwat10 Why do we need it, then?

Will this always use an empty `Uint8Array`?

https://github.com/FuelLabs/fuels-ts/blob/d0c381e53bcd3ea3528ab4accdb44fd87fadca95/packages/account/src/providers/transaction-request/transaction-request.ts#L342

https://github.com/FuelLabs/fuels-ts/blob/d0c381e53bcd3ea3528ab4accdb44fd87fadca95/packages/account/src/providers/transaction-request/transaction-request.ts#L386

I'm confused.

Reply:
The nomenclature for input coins/messages has leaked into the `Predicate` class and IMO a better naming for the `predicate` and `predicateData` properties from the user's POV is `address` (of type `string | Uint8Array`) and `data` (type depends on abi). @Dhaiwat10 sorry for opening yet another topic on this PR, but given that we're breaking stuff this is the ideal opportunity to discuss it.

Reply:
@arboleya I am confused as well. I can remove it and see if all tests pass if you want.

Reply:
@nedsalk Which `predicate` variable are you referring to here?

Reply:
@nedsalk Sorry, the `predicate` property of input coins/messages leaked into my comment as well :sweat_smile: I meant the `bytes` property of `Predicate` that is used to set the `predicate` property of input coins/messages.

Reply:
In the interest of merging this PR asap since it needs to unblock some follow-up issues

1. I created an issue for the investigation of the seemingly unset class property: https://github.com/FuelLabs/fuels-ts/issues/new
2. @nedsalk I suggest we open a discussion to discuss the nomenclature for the `Predicate` class in general?